packages/gatsby-telemetry/src/store.js

```diff
@@ -7,84 +6,42 @@ const {
   existsSync,
   unlinkSync,
 } = require(`fs`)
-const { ensureDirSync } = require(`fs-extra`)
-const Configstore = require(`configstore`)
 
 module.exports = class Store {
-  config = new Configstore(`gatsby`, {}, { globalConfigPath: true })
-
-  constructor() {}
-
-  getConfig(key) {
-    if (key) {
-      return this.config.get(key)
-    }
-    return this.config.all
-  }
-
-  updateConfig(...fields) {
-    this.config.set(...fields)
+  constructor(parentFolder) {
+    this.parentFolder = parentFolder
```

Comment:
you're only using this inside the constructor so this can be removed. (or I missed something)

Reply:
True, but we put it on the instance because it's likely that stuff in the future will need it

Reply:
I would only add it when it's necessary. Adding code that you might need in the future can bloat your codebase. I know this is a bad example as it's a one-liner.

More on this topic YAGNI:
https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it
https://martinfowler.com/bliki/Yagni.html

Reply:
Haha, okay fair enough. Removed!